addons/web_editor/static/src/js/editor/odoo-editor/src/OdooEditor.js

```diff
@@ -2414,22 +2414,19 @@ export class OdooEditor extends EventTarget {
         if (this._isResizingTable || !traversedNodes.some(node => !!closestElement(node, 'td') && !closestElement(node, '[data-oe-protected="true"]'))) {
             return false;
         }
-        let range;
-        if (selection.rangeCount > 1) {
-            // Firefox selection in table works with multiple ranges.
-            const startRange = getDeepRange(this.editable, {range: selection.getRangeAt(0)});
-            const endRange = getDeepRange(this.editable, {range: selection.getRangeAt(selection.rangeCount - 1)});
-            range = this.document.createRange();
-            range.setStart(startRange.startContainer, 0);
-            range.setEnd(endRange.startContainer, 0);
-        } else {
-            // We need the triple click correction only for a bug in firefox
-            // where it gives a selection of a full cell as tr 0 tr 1. The
-            // correction makes it so it gives us the cell and not its neighbor.
-            // In all other cases we don't want to make that correction so as to
-            // avoid flicker when hovering borders.
-            range = getDeepRange(this.editable, { correctTripleClick: anchorNode && anchorNode.nodeName === 'TR' });
+        if (this.isFirefox) {
+            if (selection.rangeCount > 1) {
+                this.historyResetLatestComputedSelection(true);
+            } else if(ev && closestElement(ev.target, 'table') === closestElement(selection.anchorNode, 'table')) {
+                setSelection(selection.anchorNode, selection.anchorOffset, ev.target, 0);
```

Comment:
We used to set a range, not a selection because we call `_handleSelectionInTable` in `_onSelectionChange` so here you're causing a risk of infinite loop! Same is true with your call to `historyResetLatestComputedSelection` above.

Reply:
I've checked that these changes do not cause issues like infinite loops, and I haven't encountered any such problems. As I mentioned before, in Firefox, the selection shifts to the `tr` when selecting mutiple `tds`, so if I don't set the selection back properly, the cursor will always come to the second `p`, as `getdeepRange` in `_applyRawCommand` will set the selection at the first `p` of the last `td` before `oEnter` gets called. 

If you believe it's not advisable to set the selection here, then another option would be to set the selection based on the selected `tds` at the time of `mouseup`. However, at that time, it's difficult to decide which `td` should be the anchor node and which should be the focus node.

Reply:
> In Firefox, the selection shifts to the tr when selecting mutiple tds

The original code here took into account the fact that Firefox supports multiple ranges on a selection. So, if I remember correctly, when selecting multiple tds, Firefox has multiple ranges that span the separate cells and rows. The original code made a new single range by taking the full span of these multiple ranges. Isn't that ok anymore?

Also, to avoid infinite loops for sure, I believe you can avoid using `setSelection` by simply creating a range for your computations rather than actually setting the document's selection.

Reply:
This doesn't address our initial issue because even if we create a new single range in the `_handleSelectionInTable` function, our selection still remains on `tr`, and the enter process occurs based on this selection. So when Enter is pressed, it triggers `applyRawCommand`, within which `getDeepRange` sets the selection to the first p of the last row, and after that, `oEnter` is executed based on this position

Reply:
I'm wondering if that's not the issue in the first place: that `oEnter` should do something special when it comes to tables so it takes the custom table selection (the classes) in consideration, and ignores the browser selection in the table. What do you think @dmo-odoo ?

Reply:
@Zynton Well I have a gut feeling that enter should behave differently in the particular case of tables, yes. The reason I think that is because I'm confused about what is supposed to happen when selecting multiple cells and press ENTER? I mean, it can't add a paragraph or anything, and it's nothing like BACKSPACE which is expected to empty the cells so... maybe it should do nothing at all? If selected multiple TDs then return ? I checked in GDocs but their behavior feels arbitrary and, frankly, buggy. cc @dhba-odoo

Reply:
That would be a decision for @pierre-pde, wouldn't it?

Reply:
Just had a discussion with Pierre. The answer is: Only the cell that was the anchor for the start of the selection should be modified, ignoring the special table selection. The content of the anchor cell should be wiped out and end up with `<p><br></p><p><br>[]</p>`.

Reply:
@Zynton As you mentioned that `oEnter` should do something special so it takes the custom table selection (the classes) rather than the browser's selection. However, with the new specs by @dmo-odoo, we considered setting the selection to the first `querySelector` value of `o_selected_td`. The issue with this approach is that if the selection is backward, the anchor node wouldn't be the first `o_selected_td`. We coundn't find another method to determine the correct starting point, so we are still reliant on the (second commit) browser's selection.

Reply:
Yes I guess with this new spec once you've determined the selection is across cells but contained within one table, you'd have to use the browser selection to determine which cell to modify (but you have to consider the whole cell selected, which the browser might not).

Reply:
Since the comment is still displayed, which means the line has not been changed, and the comment thread has not been marked resolved, I have a feeling the last comment of Zynton has not been applied, in particular the part about the whole cell being selected. That being said, when I try on runbot it seems to work. What am I missing? Because I don't understand how `setSelection(selection.anchorNode, selection.anchorOffset, ev.target, 0)` would in fact select the whole TD?

Reply:
In Firefox, multiple ranges cause the original selection to be lost. To address this issue and preserve the original selection, I have created this commit. It is not related to single cell selection.
If you are referring to the single cell where the enter process will occur, then it will be at this [line](https://github.com/odoo/odoo/pull/139312/commits/405781d5ed9e484e9d6936054e591c2c575a2d05#diff-c309d19284e2e1f38a1dba1d68e0262371686595f3668a3bc0f1ca35e3189a0bR3413) in first commit